src/main.cpp

```diff
@@ -44,32 +44,34 @@ int main(int argc, char *argv[])
 
 	glfwSetScrollCallback(window, scroll_callback);
 	Camera* camera = new Camera(glm::vec3(4,3,3), glm::vec3(0,0,0), glm::vec3(0,1,0));
-	inputManager = new InputManager(window, camera);
 
 	// Create Geometry
 	Geometry* geometry = new Geometry("models/cube/cube.obj", GL_TRIANGLES);
 
 	// Create Camera
-	// Create Scene
-	Scene scene("shaders/vertex.glsl", "shaders/fragment.glsl", camera);
-	scene.AddGeometry(geometry);
+	Shader program("shaders/vertex.glsl","shaders/fragment.glsl");
+	Shader programOutline("shaders/vertex.glsl","shaders/outline.frag");
+	Scene* scene = new Scene(&program, &programOutline, camera);
+	scene->AddGeometry(geometry);
 
-	CheckGLErrors();
+	inputManager = new InputManager(window, camera, scene);
```

Comment:
Remove new keyword from here.

inputManager = &InputManager(window, camera, scene);

InputManager should never go out of scope. If it does, program should crash.

If there is a good reason otherwise, just make sure to delete(inputManager) at the end.

Reply:
inputManager is a global pointer. Keeping it as pointer with new.